### PR TITLE
paper key input screen per mocks

### PR DESCRIPTION
**Pull request by pzduniak**
@keybase/react-hackers 
@keybase/y2ksquad

### CODE REVIEW

File: shared/provision/paper-key/index.tsx [hunk 1]
```diff
@@ -1,24 +1,45 @@
...
+            disabled: !this.state.paperKey,
+            label: 'Continue',
+            onClick: this._onSubmit,
+            type: 'Success',
+            waiting: props.waiting,
+          },
+        ]}
+        noBackground={true}
+        onBack={this.props.onBack}
+        title="Enter your paper key"
```
Comment (buoyad) [inline shared/provision/paper-key/index.tsx]: Let's try to fit this without ellipsizing
image

File: shared/chat/conversation/rekey/enter-paper-key.tsx [hunk 2]
```diff
@@ -2,13 +2,15 @@ import EnterPaperkey from '../../../provision/paper-key'
...
+import * as Constants from '../../../constants/provision'
 
 type OwnProps = {}
 
 export default Container.connect(
-  () => ({
+  state => ({
     error: '',
     hint: '',
+    waiting: Container.anyWaiting(state, Constants.waitingKey),
```
Comment (buoyad) [inline shared/chat/conversation/rekey/enter-paper-key.tsx]: Is this rekey dialog connected to the provision waiting key? If not can you change this?

File: shared/provision/paper-key/container.tsx [hunk 3]
```diff
@@ -30,8 +34,11 @@ export default Container.connect(
       error: stateProps.error,
       hint: stateProps.hint,
       onBack:
-        loggedInAccounts.length > 0 ? () => dispatchProps._onLogIn(loggedInAccounts[0] || '') : undefined,
+        loggedInAccounts.length > 0
+          ? () => dispatchProps._onLogIn(loggedInAccounts[0] || '')
+          : dispatchProps._onBack,
```
Comment (buoyad) [inline shared/provision/paper-key/container.tsx]: Have you tested this case?

Comment (pzduniak) [reply shared/provision/paper-key/container.tsx]: Which one? The user switching case wasn't changed, I'd rather not mess around with it as it's out of scope of the rework.

File: shared/chat/conversation/rekey/enter-paper-key.tsx (hunk 2 above)
Comment (pzduniak) [reply shared/chat/conversation/rekey/enter-paper-key.tsx]: woops, shouldnt have copypasted this, fixed

File: shared/provision/paper-key/container.tsx (hunk 3 above)
Comment (buoyad) [reply shared/provision/paper-key/container.tsx]: The case where you're logging in on a fresh install with a paper key and hit "back" on that screen. I think it should take you back to join-or-login

Comment (pzduniak) [reply shared/provision/paper-key/container.tsx]: Yeah, it works.

**Pull request merged**